docs/public/components/testing.md

```diff
@@ -1,3 +1,210 @@
+
+To recapitulate form the [previous page](creating-components.md). We created a component that compresses the payload before sending it to the Datadog backend. The component had twop separate implementations.
+
+This our components interface:
+
+=== ":octicons-file-code-16: comp/compression/def/component.go"
+    ```go
+    type Component interface {
+        // Compress compresses the input data.
+        Compress([]byte) ([]byte, error)
+
+        // Decompress decompresses the input data.
+        Decompress([]byte) ([]byte, error)
+    }
+    ```
+
+We must ensure the `Compress` and `Decompress` functions behave correctly. 
+
+Writing test for a component implementation follow the same rules as any other test in a Go project. Here is the documentation for the [testing](https://pkg.go.dev/testing) package.
+
+For this example we are going to write a test file for the `zstd` implementation. For that we are going to create a new file in our `impl-zstd` folder and we are going to name it `component_test.go`. Inside our test file we are going to initialize the component's dependencies, create a new component instance and test the behaviour.
+
+### Initialize the component's dependencies
+
+All components expect a `Requires` struct with all the necessary. Inside that struct we declare all our dependencies. To ensure we can create a component instance we are going to create a `requires` instance. 
+
+Our `Requires` struct declared a dependency on the config component and the logs component. The next code snippet shows how to create the `Require` struct.
+
+=== ":octicons-file-code-16: comp/compression/impl-zstd/component_test.go"
+    ```go
+    package implzstd
+    
+    import (
+      "testing"
+
+      "github.com/DataDog/datadog-agent/pkg/util/fxutil"
+
+      "github.com/DataDog/datadog-agent/comp/core/config"
+      "github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+    )
+    
+    func TestCompress(t *testing.T) {
+      logComponent := fxutil.Test[log.Component](t, logimpl.MockModule())
+      configComponent := fxutil.Test[config.Component](t, config.MockModule())
+      
+      requires := Requires{
+        Conf: configComponent,
+        Log: logComponent,
+      }
+      // [...]
+    }
+    ```
+    
+To create the log and config component from their respective mocks we used `fxutil.Test[T]` generic function. This functions make sure to initialize any Fx dependency needed for the Mock and return a instance of `T`.
+
+!!! warning "is not intended to use `fxutil` package inside the implementation"
+    We are migrating from a previous file structure to the one outlined in the [creating a component page](creating-components.md#file-hierarchy). For now it is ok to use as it allow us to migrate one component at a time. in the future there we would be restrciting the use Fx related code inside the implementation folder using a custom Go linter. 
```

Comment:
I think our example should show the correct way and then show how to handle legacy component. We should put the accent on the correct way to do this.